runtime/lua/vim/lsp/completion.lua

```diff
@@ -250,9 +251,18 @@ function M._lsp_to_complete_items(result, prefix, client_id)
   end
 
   local candidates = {}
+  local bufnr = api.nvim_get_current_buf()
   for _, item in ipairs(items) do
     if matches(item) then
       local word = get_completion_word(item)
+      local abbr = item.label
+      if buf_handles[bufnr].abbr_format then
+        if abbr then
+          abbr = buf_handles[bufnr].abbr_format(abbr)
+        else
+          word = buf_handles[bufnr].abbr_format(word)
+        end
+      end
```

Comment:
`item.label` is afaik a mandatory property so the `if abbr then` case should always be true and the else brach would be dead code.


I'm overall not sure about this. Seems a bit too specific. If servers report long labels maybe it should be brought up with them.

Or otherwise another option could be to set a `convert(item)` function that can return the full structure, including word, kind, info, etc.. ` and also apply a `tbl_extend("keep", ...)` on the result to fill in missing properties.

This way you could customize `abbr` as well as any of the other properties.

Reply:
> I'm overall not sure about this. Seems a bit too specific. If servers report long labels maybe it should be brought up with them.

lsp no universal standard  here.  our api functions luals also return function name with all parameters.

convert is a more general way.